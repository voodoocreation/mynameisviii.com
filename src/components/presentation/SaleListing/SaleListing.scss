@import "../../../scss/variables", "../../../scss/mixins";

.SaleListing {
  padding: $spacing-half;

  h3 {
    margin: 0;
  }

  > * {
    display: flex;
    align-items: center;
  }

  &--icon {
    display: flex;
    align-items: center;
    font-size: 32px;
  }

  &--details {
    padding: 0 $spacing-half;
  }

  &--price {
    font-size: 32px;
  }

  @include when-above(1400px) {
    &--icon {
      font-size: 64px;
    }
  }
}
