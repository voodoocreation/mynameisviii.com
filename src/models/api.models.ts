import { arrayToAssoc } from "../transformers/transformData";

interface IWithSlug {
  slug: string;
}

export type TLastEvaluatedKey<K extends string> = Record<
  K | "isActive" | "slug",
  string
>;

export interface IRawDynamoResponse<T extends {}, K extends string> {
  items: T[];
  lastEvaluatedKey?: TLastEvaluatedKey<K>;
}

export interface IDynamoResponse<T extends IWithSlug, K extends string> {
  readonly items: Record<string, T>;
  readonly lastEvaluatedKey?: TLastEvaluatedKey<K>;
}

export interface IRawS3Response<T extends {}> {
  isTruncated?: boolean;
  items: T[];
}

export interface IS3Response<T extends IWithSlug> {
  readonly isTruncated: boolean;
  readonly items: Record<string, T>;
}

export const dynamoResponse = <T extends IWithSlug, K extends string>(
  options: IRawDynamoResponse<T, K>
): IDynamoResponse<T, K> => ({
  items: arrayToAssoc(options.items, "slug"),
  lastEvaluatedKey: options.lastEvaluatedKey
});

export const s3Response = <T extends IWithSlug>(
  options: IRawS3Response<T>
): IS3Response<T> => ({
  isTruncated: !!options.isTruncated,
  items: arrayToAssoc(options.items, "slug")
});
