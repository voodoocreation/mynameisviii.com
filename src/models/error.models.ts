export interface IError {
  readonly message: string;
  readonly status: number;
}

export const error = (options: Partial<IError> = {}): IError => ({
  message: options.message || "An error has occurred",
  status: options.status || 500,
});
