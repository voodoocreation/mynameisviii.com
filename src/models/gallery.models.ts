import dayjs from "dayjs";

export interface IRawGallery {
  description?: string;
  imageUrl?: string;
  images?: Array<Partial<IGalleryImage>>;
  modifiedAt?: string;
  slug?: string;
  title?: string;
}

export interface IGallery {
  readonly description?: string;
  readonly imageUrl: string;
  readonly images?: IGalleryImage[];
  readonly modifiedAt: string;
  readonly slug: string;
  readonly title: string;
}

export interface IGalleryImage {
  readonly imageUrl: string;
  readonly modifiedAt: string;
}

export const galleryImage = (
  options: Partial<IGalleryImage> = {}
): IGalleryImage => ({
  imageUrl: options.imageUrl || "",
  modifiedAt: options.modifiedAt || dayjs().toISOString()
});

export const gallery = (options: IRawGallery = {}): IGallery => ({
  description: options.description,
  imageUrl: options.imageUrl || "",
  images: options.images ? options.images.map(galleryImage) : undefined,
  modifiedAt: options.modifiedAt || dayjs().toISOString(),
  slug: options.slug || "",
  title: options.title || ""
});
