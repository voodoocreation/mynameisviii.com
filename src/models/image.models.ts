export interface IImage {
  readonly imageUrl: string;
  readonly title: string;
}

export const image = (options: Partial<IImage> = {}): IImage => ({
  imageUrl: options.imageUrl || "",
  title: options.title || ""
});
