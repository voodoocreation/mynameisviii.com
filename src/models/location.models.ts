import { TYPE } from "../constants/location.constants";

export interface ILatLng {
  readonly lat: number;
  readonly lng: number;
}

export interface ILocation {
  readonly address?: string;
  readonly latLng?: ILatLng;
  readonly name: string;
  readonly type: TYPE;
  readonly url?: string;
}

export const location = (options: Partial<ILocation> = {}): ILocation => ({
  address: options.address,
  latLng: options.latLng,
  name: options.name || "",
  type: options.type || TYPE.MUSIC_VENUE,
  url: options.url,
});
