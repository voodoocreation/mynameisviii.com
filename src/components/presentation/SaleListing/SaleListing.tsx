import * as React from "react";
import { FaTicketAlt } from "react-icons/fa";
import { MdLanguage } from "react-icons/md";

import { extractDomain } from "../../../helpers/dataTransformers";
import { IOffer } from "../../../models/offer.models";
import Link from "../Link/Link";
import Meta from "../Meta/Meta";
import Price from "../Price/Price";

import "./SaleListing.scss";

const SaleListing: React.FC<IOffer> = ({ ...sale }) => (
  <article className="SaleListing">
    <Link className="SaleListing--link" href={sale.url} isExternal>
      <div className="SaleListing--icon">
        <FaTicketAlt />
      </div>

      <div className="SaleListing--details">
        <h3>
          {sale.name}:{" "}
          <Price currency={sale.priceCurrency} value={sale.price} />
        </h3>

        <section className="SaleListing--meta">
          {sale.url ? (
            <Meta
              className="SaleListing--website"
              icon={<MdLanguage />}
              labelIntlId="WEBSITE"
            >
              {extractDomain(sale.url)}
            </Meta>
          ) : null}
        </section>
      </div>
    </Link>
  </article>
);

export default SaleListing;
