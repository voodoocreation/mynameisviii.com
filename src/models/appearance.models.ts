import dayjs from "dayjs";

import { BOOLEAN } from "../constants/api.constants";
import { STATUS, TYPE } from "../constants/appearance.constants";
import { IImage, image } from "./image.models";
import { ILocation, location } from "./location.models";
import { IOffer, offer } from "./offer.models";
import { IOrganization, organization } from "./organization.models";
import { IPerformer, IRawPerformer, performer } from "./performer.models";

export interface IRawAppearance {
  acts?: Array<Partial<IRawPerformer>>;
  audience?: string;
  description?: string;
  finishingAt?: string;
  imageUrl?: string;
  images?: IImage[];
  isActive?: BOOLEAN;
  location?: Partial<ILocation>;
  ogImageUrl?: string;
  organizer?: Partial<IOrganization>;
  rsvpUrl?: string;
  sales?: Array<Partial<IOffer>>;
  slug?: string;
  startingAt?: string;
  status?: STATUS;
  title?: string;
  type?: TYPE;
}

export interface IAppearance {
  readonly acts: IPerformer[];
  readonly audience?: string;
  readonly description: string;
  readonly finishingAt: string;
  readonly imageUrl: string;
  readonly images: IImage[];
  readonly isActive: boolean;
  readonly location: ILocation;
  readonly ogImageUrl: string;
  readonly organizer: IOrganization;
  readonly rsvpUrl?: string;
  readonly sales: IOffer[];
  readonly slug: string;
  readonly startingAt: string;
  readonly status: STATUS;
  readonly title: string;
  readonly type: TYPE;
}

export const appearance = (options: IRawAppearance = {}): IAppearance => ({
  acts: options.acts ? options.acts.map(performer) : [],
  audience: options.audience,
  description: options.description || "",
  finishingAt: options.finishingAt || dayjs().toISOString(),
  imageUrl: options.imageUrl || "",
  images: options.images ? options.images.map(image) : [],
  isActive: (options.isActive && options.isActive === BOOLEAN.TRUE) || false,
  location: location(options.location),
  ogImageUrl: options.ogImageUrl || "",
  organizer: organization(options.organizer),
  rsvpUrl: options.rsvpUrl,
  sales: options.sales ? options.sales.map(offer) : [],
  slug: options.slug || "",
  startingAt: options.startingAt || dayjs().toISOString(),
  status: options.status || STATUS.SCHEDULED,
  title: options.title || "",
  type: options.type || TYPE.MUSIC,
});
