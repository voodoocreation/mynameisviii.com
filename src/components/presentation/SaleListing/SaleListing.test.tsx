import { offer } from "../../../models/root.models";
import WrapperWithIntl from "../../../utilities/WrapperWithIntl";
import SaleListing from "./SaleListing";

const component = new WrapperWithIntl(SaleListing).withDefaultProps(
  offer({
    name: "Sale",
    price: 99.99,
    priceCurrency: "USD",
  })
);

describe("[presentation] <SaleListing />", () => {
  describe("when url isn't defined", () => {
    const wrapper = component.render();

    it("doesn't render the website meta item", () => {
      expect(wrapper.find(".SaleListing--website")).toHaveLength(0);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when url is defined", () => {
    const wrapper = component
      .withProps({
        url: "https://ticket.com/id",
      })
      .render();

    it("renders the website meta item", () => {
      expect(wrapper.find(".SaleListing--website")).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
