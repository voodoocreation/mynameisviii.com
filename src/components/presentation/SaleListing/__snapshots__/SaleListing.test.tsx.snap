// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <SaleListing /> when url is defined matches snapshot 1`] = `
<article
  class="SaleListing"
>
  <a
    class="SaleListing--link"
    href="https://ticket.com/id"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="SaleListing--icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 160h320v192H128V160zm400 96c0 26.51 21.49 48 48 48v96c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48v-96c26.51 0 48-21.49 48-48s-21.49-48-48-48v-96c0-26.51 21.49-48 48-48h480c26.51 0 48 21.49 48 48v96c-26.51 0-48 21.49-48 48zm-48-104c0-13.255-10.745-24-24-24H120c-13.255 0-24 10.745-24 24v208c0 13.255 10.745 24 24 24h336c13.255 0 24-10.745 24-24V152z"
        />
      </svg>
    </div>
    <div
      class="SaleListing--details"
    >
      <h3>
        Sale: 
        <span
          class="Price"
        >
          US$99.99
        </span>
      </h3>
      <section
        class="SaleListing--meta"
      >
        <div
          class="Meta SaleListing--website"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
           
          <span
            class="Meta--label"
          >
            Website:
          </span>
           
          <span
            class="Meta--value"
          >
            ticket.com
          </span>
        </div>
      </section>
    </div>
  </a>
</article>
`;

exports[`[presentation] <SaleListing /> when url isn't defined matches snapshot 1`] = `
<article
  class="SaleListing"
>
  <span
    class="SaleListing--link"
  >
    <div
      class="SaleListing--icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 160h320v192H128V160zm400 96c0 26.51 21.49 48 48 48v96c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48v-96c26.51 0 48-21.49 48-48s-21.49-48-48-48v-96c0-26.51 21.49-48 48-48h480c26.51 0 48 21.49 48 48v96c-26.51 0-48 21.49-48 48zm-48-104c0-13.255-10.745-24-24-24H120c-13.255 0-24 10.745-24 24v208c0 13.255 10.745 24 24 24h336c13.255 0 24-10.745 24-24V152z"
        />
      </svg>
    </div>
    <div
      class="SaleListing--details"
    >
      <h3>
        Sale: 
        <span
          class="Price"
        >
          US$99.99
        </span>
      </h3>
      <section
        class="SaleListing--meta"
      />
    </div>
  </span>
</article>
`;
